common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java

```diff
@@ -74,47 +79,61 @@ public static String modifyMultipleMatchKeyword(String query){
                     List<Predicate> predicates = new ArrayList<>();
 
                     // Check that the concept name, code, or ID matches the query string.
-                    List<Predicate> queryPredicates = new ArrayList<>();
-                    Expression<Double> matchExp = criteriaBuilder.function("match", Double.class,
-                            root.get("conceptName"), criteriaBuilder.literal(keyword));
-                    queryPredicates.add(criteriaBuilder.greaterThan(matchExp, 0.0));
-                    queryPredicates.add(criteriaBuilder.equal(root.get("conceptCode"),
+                    List<Predicate> conceptCode_Id = new ArrayList<>();
+
+                    conceptCode_Id.add(criteriaBuilder.equal(root.get("conceptCode"),
                             criteriaBuilder.literal(query)));
+
                     try {
                         long conceptId = Long.parseLong(query);
-                        queryPredicates.add(criteriaBuilder.equal(root.get("conceptId"),
+                        conceptCode_Id.add(criteriaBuilder.equal(root.get("conceptId"),
                                 criteriaBuilder.literal(conceptId)));
                     } catch (NumberFormatException e) {
                         // Not a long, don't try to match it to a concept ID.
                     }
-                    predicates.add(criteriaBuilder.or(queryPredicates.toArray(new Predicate[0])));
+
+                    List<Predicate> conceptName_Filter = new ArrayList<>();
+
+
+                    Expression<Double> matchExp = criteriaBuilder.function("match", Double.class,
+                            root.get("conceptName"), criteriaBuilder.parameter(String.class , "keyword"));
+                    conceptName_Filter.add(criteriaBuilder.greaterThan(matchExp, 0.0));
+
 
                     // Optionally filter on standard concept, vocabulary ID, domain ID
                     if (standardConceptFilter.equals(StandardConceptFilter.STANDARD_CONCEPTS)) {
```

Comment:
I'm wondering if we should get rid of this code... given the logic where we return non-standard concepts that match code or ID unconditionally, it's not clear to me that filtering on standard vs. non-standard concepts really makes sense anymore. (It's weird to explicitly ask for standard concepts in the requests, and return non-standard ones anyway...)

Reply:
No, just get rid of the filter entirely.

(We'd start out by keeping it in the API but ignoring it... and then remove it from the API.)

Reply:
But we need to only get standard concepts . Are you saying just return standard concepts unless it is a code or concept_id match ?

Reply:
Yep, that's what I'm suggesting. We could add some option back in later when we have requirements for fetching non-standard concepts on something else.

If you prefer we can keep this filter around for future use, but then we ought to have requests that ask for standard concepts get *only* standard concepts, non-standard get only non-standard and match on everything, etc.

Reply:
I just can't predict all the use cases , but to me , be able to specify the standard_concept flag you want in a concept search seems like not a bad thing to keep around.  Especially since we already wrote all the code for it.  I understand how we display concepts in hierarchies to users and such is up in the air .... Just saying -- if they say "Databrowser team : make a button to toggle all vs standard on the search results ... "   I can do that now. 

So i would say that we make the default "S" and allow you to pass "C" "null" .

If we want to clarify the api , we could make a option for search type:  "MatchCode && MatchConceptID&&MatchConceptName"   would specify the order and what fields you wanted to search on. "MatchConceptName" would just do the concept name search. MatchCode or Concept implies that the standard_concept flag is irrelevant because it is in these contexts.   I prefer that over deleting the S option .

Reply:
Right now it is (MatchCode) or (MatchId) or (MatchName and standardconceptfilter). So may be it is fine like this for now. We can use the filter as how we want.

Reply:
I was talking about adding a parameter to the SearchConceptRequest object . It would be an optional array with the match types you wanted. The array order sets the precedence. It would default to [Code, ConceptId , Name] ..   But if you  did want only standard concepts you would probably be searching on name.  That option only makes sense searching on name.  Unless a couple vocabularies use the same code and you would get more than one. It is rare in any case and there won't be enough matches to create noise imo. 

We could also do : matchType: [ {type:Code, standard_concept:false},{type:ConceptId, standard_concept:false, {type:Name, standard_concept:true} ]  -- so instead of array of scalars have array of objects and put the standard_concept filter in each match type object.

Reply:
That said, its a lot of work for something we aren't using now. But given we can filter on domain, vocabulary, seems we should filter on standard_concept too just to be thorough.  That said, given that the standard_concept filter only makes sense in a concept_name search, i think we are fine as we are.  But Dan if you really don't like it , i'm fine taking it out for now until we need it.

Reply:
I think the best option is to add another StandardConceptFilter.StandardOrCodeIdMatch ... Seems a good use of the StandardConceptFilter enumerator.

Reply:
I'm OK with that, but then for the other options (standard, non-standard) we should return only those concepts... we shouldn't do ID matching for the other kind of concepts in those cases.